[cast] Multicall new (#98)
foundry-rs/starknet-foundry: docs/src/appendix/cast/multicall.md

contract_address = "map_contract"
function = "put"
inputs = ["0x123", "234"]
```

## `new`

> Generates an empty template for the multicall `.toml` file that may be later use with the `run` subcommand.

Reviewer: let's add an info, that by default it is being printed to stdout here

Author: [address dox fdbck](https://github.com/foundry-rs/starknet-foundry/pull/98/commits/722855d7bb207da35ea14576b497377f59274d74)

### `--output_path, -p <PATH>`
Optional.

> When provided, it specifies where the template should be saved. If omitted, the template contents are going to be printed out to the stdout.

Reviewer: ```suggestion
Specifies the path where the template will be saved.
```

btw, does this specify path to the directory or the file?

Author: What's wrong with the sentence that is there already? 

> btw, does this specify path to the directory or the file?

it has to be file

Reviewer: it's unnecessarily long. Docs should be concise, but informative.

> it has to be file
cool, then let's state it explicitly, that this is a path to the file to which template will be saved.

Author: > it's unnecessarily long. Docs should be concise, but informative.

Ok, but I do not think the line you suggested is good. This is because it is missing what is said in the original one (what happens when you provide this arg and what happens if you do not).

I also don't think the original line is botheringly long.

Author: > cool, then let's state it explicitly, that this is a path to the file to which template will be saved.

[dox fdbck](https://github.com/foundry-rs/starknet-foundry/pull/98/commits/3a6034e60aabceb786be6e99c41badae1fb8543a)

Author: [updt](https://github.com/foundry-rs/starknet-foundry/pull/98/commits/222bb30e6236945f904fd79a0cc66021efc637f2)

### `--overwrite, -o <OVERWRITE>`
Optional.

> If the `--output-path` argument is provided and such a file already exists, this flag decides whether it should be overwritten or not.

Reviewer: ```suggestion
Overwrites the file specified by `--output-path`, if it exists.
```

Author: I think the original sentence's better, in the suggested one, in the latter part, "it" is confusing - it is not clear what it refers to (either it's the file or the output path).

Besides, I think the original sentence is easier to understand.

contract_address = "map_contract"
function = "put"
inputs = ["0x123", "234"]
```

## `new`

> Generates an empty template for the multicall `.toml` file that may be later use with the `run` subcommand. It either outputs it to a new file or to the standard output.

Reviewer: ```suggestion
Generates an empty template for the multicall `.toml` file that may be later used with the `run` subcommand. It either outputs it to a new file or to the standard output.
```

Author: [tense fix](https://github.com/foundry-rs/starknet-foundry/pull/98/commits/d89ef108744ccdcc0001798a19daf940079083e6)

> ### `--output_path, -p <PATH>`

Reviewer: ```suggestion
### `--output-path, -p <PATH>`
```

Author: [underscore to dash](https://github.com/foundry-rs/starknet-foundry/pull/98/commits/a89cfa5029992b32f6e23d35afa21d6ef1a7485f)

Optional.

> When provided, it specifies where the template should be saved, it has to be a file path.
> If omitted, the template contents are going to be printed out to the stdout.

Reviewer: ```suggestion
Specifies a file path where the template should be saved.
If omitted, the template contents will be printed out to the stdout.
```

Author: ok what's wrong with 

> When provided, it specifies

?
As for 

> it has to be a file path

you yourself said we should mention this explicitly [here](https://github.com/foundry-rs/starknet-foundry/pull/98#discussion_r1268026976).

Author: [are going to => will](https://github.com/foundry-rs/starknet-foundry/pull/98/commits/70be9ff5d0ce5e0b497e0cc9d686e59f0df9dacd)

Author: > The diff I propose states that explicitly - `Specifies a file path` - and not `Specifies a path`.

ok fair I missed that

Author: [updt](https://github.com/foundry-rs/starknet-foundry/pull/98/commits/222bb30e6236945f904fd79a0cc66021efc637f2)

### `--overwrite, -o <OVERWRITE>`
Optional.

> Overwrites the file specified by `--output-path`, if it exists.

Reviewer: I thought that we agreed on different wording, to ensure noone thinks 'it' refers to the flag (but the file)?
https://github.com/foundry-rs/starknet-foundry/pull/98#discussion_r1268035087

Author: [fix dox](https://github.com/foundry-rs/starknet-foundry/pull/98/commits/cc78ee39dbe5a143333953db5261d46c6cce69b8)
